optaplanner-core/src/main/java/org/optaplanner/core/config/heuristic/selector/entity/pillar/SubPillarConfigPolicy.java

```diff
@@ -0,0 +1,106 @@
+package org.optaplanner.core.config.heuristic.selector.entity.pillar;
+
+import java.util.Comparator;
+import java.util.Optional;
+
+public final class SubPillarConfigPolicy {
+
+    private final boolean subPillarEnabled;
+    private final int minimumSubPillarSize;
+    private final int maximumSubPillarSize;
+    private final Comparator<?> entityComparator;
+
+    private SubPillarConfigPolicy(int minimumSubPillarSize, int maximumSubPillarSize) {
+        this.subPillarEnabled = true;
+        this.minimumSubPillarSize = minimumSubPillarSize;
+        this.maximumSubPillarSize = maximumSubPillarSize;
+        validateSizes();
+        this.entityComparator = null;
+    }
+
+    private SubPillarConfigPolicy(int minimumSubPillarSize, int maximumSubPillarSize, Comparator<?> entityComparator) {
+        this.subPillarEnabled = true;
+        this.minimumSubPillarSize = minimumSubPillarSize;
+        this.maximumSubPillarSize = maximumSubPillarSize;
+        validateSizes();
+        if (entityComparator == null) {
+            throw new IllegalStateException("entityComparator must not be null.");
```

Comment:
error messages start with a capital
error messages are sentances so they have "The" and "A" etc.
=> "The entityComparator must not be null.")